File: analytics/src/test/java/io/appmetrica/analytics/ReceiptTest.java
package io.appmetrica.analytics;

import io.appmetrica.analytics.testutils.CommonTest;
import org.assertj.core.api.SoftAssertions;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

@RunWith(RobolectricTestRunner.class)
public class ReceiptTest extends CommonTest {

    @Test
    public void testOptional() {
        Revenue.Receipt receipt = Revenue.Receipt.newBuilder()
                .withData("data")
                .withSignature("signature").build();
        SoftAssertions softAssertion = new SoftAssertions();
        softAssertion.assertThat(receipt.data).as("data").isEqualTo("data");
        softAssertion.assertThat(receipt.signature).as("signature").isEqualTo("signature");
        softAssertion.assertAll();
    }

}

File: analytics/src/test/java/io/appmetrica/analytics/ReceiptTest.kt
package io.appmetrica.analytics

import io.appmetrica.analytics.testutils.CommonTest
import org.assertj.core.api.SoftAssertions
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class ReceiptTest : CommonTest() {

    @Test
    fun optional() {
        val receipt = Revenue.Receipt.newBuilder()
            .withData("data")
            .withSignature("signature").build()
        val softAssertion = SoftAssertions()
        softAssertion.assertThat(receipt.data).describedAs("data").isEqualTo("data")
        softAssertion.assertThat(receipt.signature).describedAs("signature").isEqualTo("signature")
        softAssertion.assertAll()
    }
}
